paddle/operators/recurrent_op.h

```diff
@@ -117,10 +130,14 @@ class RecurrentOp final : public framework::OperatorBase {
     alg_.Run(scope, dev_ctx);
   }
 
+  void set_stepnet(std::shared_ptr<NetOp> net) { stepnet_ = net; }
+  const NetOp* stepnet() const { return stepnet_.get(); }
+
   static const rnn::ArgumentName kArgName;
 
  private:
   RecurrentAlgorithm alg_;
+  std::shared_ptr<NetOp> stepnet_;
```

Comment:
In `pybind.cc`, a `std::shared_ptr` is passed to `RecurrentOp` as an argument, so here is a shared pointer too.